src/Service/QuestionInstanceService.php

```diff
@@ -74,8 +73,23 @@ public function createQuestionInstances(QuizTemplate $quizTemplate, QuizInstance
         }
     }
 
-    public function getQuestionInstance(int $id)
+    public function getQuestionInstance(int $id): QuestionInstance
+    {
+        /** @var QuestionInstance $question */
+        $question = $this->questionInstanceRepo->find($id);
+
+        return $question;
+    }
+
+    public function getAllQuestionsForQuizInstance(int $quizInstanceId): array
     {
-        return $this->questionInstanceRepo->find($id);
+        $questions=$this->questionInstanceRepo->getQuestions($quizInstanceId);
+        $answers=[];
+        foreach ($questions as $question){
+            $answers[]=$this->answerInstanceRepo->getAnswers($question->getId());
```

Comment:
Use question ID as key.
If there is time - Pair object that pairs questions and answers.